File: Linear/Dynamic/Linked List/Doubly Linked List/01_Traversing/Backward/Iterative/Traversing.py
class Node:
    def __init__(self, data):
        self.data = data
        self.prev = None
        self.next = None

"""Function to traverse the doubly
linked list in backward direction"""
def backwardTraversal(tail):
    while tail is not None:
        print(tail.data, end="")
        if tail.prev is not None:
            print(" <-> ", end="")
        tail = tail.prev
    print()

# Create a hardcoded doubly linked list:
# 1 <-> 2 <-> 3
head = Node(1)
second = Node(2)
third = Node(3)

head.next = second
second.prev = head
second.next = third
third.prev = second

print("Backward Traversal: ", end="")
backwardTraversal(third)

File: Linear/Dynamic/Linked List/Doubly Linked List/01_Traversing/Backward/Iterative/Traversing.java
class Node {
    int data;
    Node prev;
    Node next;

    Node(int data) {
        this.data = data;
        this.prev = null;
        this.next = null;
    }
}

class IterativeBackwardTraversing {
    /*Function to traverse the doubly
    linked list in backward direction*/
    public static void backwardTraversal(Node tail) {
        while (tail != null) {
            System.out.print(tail.data);
            if (tail.prev != null)
            System.out.print(" <-> ");
            tail = tail.prev;
        }

        System.out.println();
    }

    public static void main(String[] args) {
        // Create a hardcoded doubly linked list:
        // 1 <-> 2 <-> 3
        Node head = new Node(1);
        Node second = new Node(2);
        Node third = new Node(3);

        head.next = second;
        second.prev = head;
        second.next = third;
        third.prev = second;

        System.out.print("Backward Traversal: ");
        backwardTraversal(third);
    }
}